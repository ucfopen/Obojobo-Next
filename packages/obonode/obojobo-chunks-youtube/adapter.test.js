jest.mock('obojobo-document-engine/src/scripts/common/models/obo-model', () => {
	return require('obojobo-document-engine/__mocks__/obo-model-adapter-mock').default
})
import OboModel from 'obojobo-document-engine/src/scripts/common/models/obo-model'

import YoutubeAdapter from './adapter'

describe('Youtube adapter', () => {
	test('construct builds without attributes', () => {
		const model = new OboModel({})
		YoutubeAdapter.construct(model)
		expect(model.modelState).toMatchSnapshot()
	})

	test('construct builds with attributes', () => {
		const attrs = {
			content: {
				videoId: 'example_video_id',
				videoUrl: 'https://www.youtube.com/watch?v=example_video_id',
				startTime: 5,
				endTime: 10
			}
		}
		const model = new OboModel(attrs)
		YoutubeAdapter.construct(model, attrs)
		expect(model.modelState).toMatchSnapshot()
	})

	test('clone creates a copy', () => {
		const attrs = {
			content: {
				videoId: 'example_video_id',
				videoUrl: 'https://www.youtube.com/watch?v=example_video_id',
				startTime: 5,
				endTime: 10
			}
		}
		const a = new OboModel(attrs)
		const b = new OboModel({})

		YoutubeAdapter.construct(a, attrs)
		YoutubeAdapter.clone(a, b)

		expect(a.modelState).not.toBe(b.modelState)
		expect(a.modelState).toEqual(b.modelState)
	})

	test('toJSON builds a JSON representation', () => {
		const json = { content: {} }
		const attrs = {
			content: {
				videoId: 'example_video_id',
				videoUrl: 'https://www.youtube.com/watch?v=example_video_id',
				startTime: 5,
				endTime: 10
			}
		}
		const model = new OboModel(attrs)

		YoutubeAdapter.construct(model, attrs)
		YoutubeAdapter.toJSON(model, json)

		expect(json).toMatchSnapshot()
	})
})
