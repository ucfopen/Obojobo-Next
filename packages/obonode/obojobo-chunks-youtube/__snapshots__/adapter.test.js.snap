// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Youtube adapter construct builds with attributes 1`] = `
Object {
  "endTime": 10,
  "startTime": 5,
  "videoId": "example_video_id",
}
`;

exports[`Youtube adapter construct builds without attributes 1`] = `
Object {
  "endTime": null,
  "startTime": null,
  "videoId": null,
}
`;

exports[`Youtube adapter toJSON builds a JSON representation 1`] = `
Object {
  "content": Object {
    "endTime": 10,
    "startTime": 5,
    "videoId": "example_video_id",
  },
}
`;
